bindgen/callbacks.rs

```diff
@@ -231,15 +232,28 @@ pub struct DeriveInfo<'a> {
     pub kind: TypeKind,
 }
 
-/// Relevant information about a type to which new attributes will be added using
+/// Relevant information about an item to which new attributes will be added using
 /// [`ParseCallbacks::add_attributes`].
 #[derive(Debug)]
 #[non_exhaustive]
 pub struct AttributeInfo<'a> {
-    /// The name of the type.
+    /// The name of the item.
     pub name: &'a str,
-    /// The kind of the type.
-    pub kind: TypeKind,
+    /// The kind of the item.
+    pub kind: AttributeItemKind,
+}
+
+#[derive(Debug, Clone, Copy, PartialEq, Eq)]
+/// The kind of the current item.
+pub enum AttributeItemKind {
+    /// The item is a Rust `struct`.
+    Struct,
+    /// The item is a Rust `enum`.
+    Enum,
```

Comment:
Is the callback only called on the main enum / struct / union definition, or also for each of the struct fields / enum variants? For example, one might want to feature guard certain struct fields.

Reply:
Only called on enums I think, they can be implemented in different ways so I think it might be a bit harder to do. But we could have a `Constant` callback perhaps?

Reply:
Resolving this for now because
- I've rebased the branch
- The changes you've proposed to enums have to be taken into consideration